File: murnana/Unity-TestDrivenDevelopmentByExample/Assets/TestDrivenDevelopmentByExample/Scripts/Runtime/Dollar.cs
namespace TDD
{
    public sealed class Dollar : Money
    {
        public Dollar(int amount, string currency)
            : base (amount: amount, currency: currency)
        {
        }

        public override Money Times(int multiplier)
        {
            return new Dollar (amount: m_Amount * multiplier, currency: m_Currency);
        }
    }
}

File: kalen/dollar.py
from money import Money
class Dollar(Money):
    def __init__(self, amount):
        super().__init__(amount)
        self.__amount = amount
    

    def times(self, multiplier: int):
        return Dollar(self.amount * multiplier)
    
    def __eq__(self, object: Money) -> bool:
        money = object
        return self.amount == money.amount

